File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/http/server/stream/StreamControllerSpec.kt
package io.micronaut.docs.http.server.stream

import io.micronaut.context.ApplicationContext
import io.micronaut.http.HttpRequest
import io.micronaut.http.MediaType
import io.micronaut.http.client.HttpClient
import io.micronaut.runtime.server.EmbeddedServer
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class StreamControllerSpec {

    lateinit var ctx: ApplicationContext
    lateinit var client: HttpClient

    @BeforeEach
    fun setup() {
        val server = ApplicationContext.run(
            EmbeddedServer::class.java,
            mapOf(
                "myapp.updatedAt" to mapOf( // <1>
                    "day" to 28,
                    "month" to 10,
                    "year" to 1982
                )
            )
        )
        ctx = server.applicationContext
        client = ctx.createBean(HttpClient::class.java, server.url)
    }

    @AfterEach
    fun teardown() {
        ctx.close()
    }


    @Test
    fun testReceivingAStream() {
        val response: String = client.toBlocking().retrieve(
            HttpRequest.GET<Any>("/stream/write"),
            String::class.java
        )

        Assertions.assertEquals("test", response)
    }

    @Test
    fun testReturningAStream() {
        val body = "My body"
        val response = client.toBlocking().retrieve(
                HttpRequest.POST("/stream/read", body)
                        .contentType(MediaType.TEXT_PLAIN_TYPE), String::class.java)

        Assertions.assertEquals(body, response)
    }

}

File: test-suite-python/src/test/python/micronaut/docs/http/server/stream/StreamControllerSpec.py
from typing import Annotated

import java
from jakarta.inject import Inject
from micronaut.http import MediaType
from micronaut.http.client import HttpClient
from micronaut.http.client.annotation import Client
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

HttpRequest = java.type("io.micronaut.http.HttpRequest")
String = java.type("java.lang.String")


@MicronautTest
class StreamControllerSpec:
    client: Annotated[HttpClient, Inject, Client("/")]

    @Test
    def test_receiving_a_stream(self):
        response = self.client.toBlocking().retrieve(HttpRequest.GET("/stream/write"), String)

        assert response == "test"

    @Test
    def test_returning_a_stream(self):
        body = "My body"
        response = self.client.toBlocking().retrieve(
            HttpRequest.POST("/stream/read", body).contentType(MediaType.TEXT_PLAIN_TYPE),
            String,
        )

        assert response == body
